src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/check_health.py

```diff
@@ -0,0 +1,266 @@
+
+
+def print_error(message):
+    from colorama import Fore, Style, init
+    init()
+    print(Fore.RED + str(message) + Style.RESET_ALL)
+
+
+def _handle_error(error, ignore_errors):
+    if ignore_errors:
+        print_error(error.get_error_message())
+    else:
+        raise CLIError(error.get_error_message(FAQ_MESSAGE))
+
+
+def _subprocess_communicate(command_parts, shell=False):
+    from subprocess import PIPE, Popen, CalledProcessError
+    output, stderr = "", ""
+
+    try:
+        p = Popen(command_parts, stdout=PIPE, stderr=PIPE, shell=shell)
+        output, stderr = p.communicate()
+        output = output.decode('UTF-8').rstrip()
+        stderr = stderr.decode('UTF-8').rstrip()
+    except CalledProcessError as e:
+        stderr = str(e)
+
+    if stderr:
+        stderr = "Failed to run command '{}'. {}".format(
+            ' '.join(command_parts),
+            stderr
+        )
+    return output, stderr
+
+
+# Checks for the environment
+
+
+# Checks docker command, docker daemon, docker version and docker pull
+def _get_docker_status_and_version(ignore_errors, yes):
+    from ._errors import DOCKER_DAEMON_ERROR, DOCKER_PULL_ERROR, DOCKER_VERSION_ERROR
+
+    # Docker command and docker daemon check
+    docker_command, error = get_docker_command(is_diagnostics_context=True)
+    docker_daemon_available = True
+
+    if error:
+        _handle_error(error, ignore_errors)
+        if error.error_title != DOCKER_DAEMON_ERROR.error_title:
+            return  # We cannot proceed if the error is unexpected or with docker command
+        else:
+            docker_daemon_available = False
+
+    if docker_daemon_available:
+        print("Docker daemon status: available")
+
+    # Docker version check
+    output, stderr = _subprocess_communicate([docker_command, "--version"])
+    if stderr:
+        _handle_error(DOCKER_VERSION_ERROR.append_error_message(stderr), ignore_errors)
+    else:
+        print("Docker version: {}".format(output))
+
+    # Docker pull check - only if docker daemon is available
+    if docker_daemon_available:
+        if not yes:
+            from knack.prompting import prompt_y_n
+            confirmation = prompt_y_n("This will pull the image {}. Proceed?".format(IMAGE))
+            if not confirmation:
+                print_warning("Skipping pull check.")
+                return
+
+        output, stderr = _subprocess_communicate([docker_command, "pull", IMAGE])
+
+        if stderr:
+            _handle_error(DOCKER_PULL_ERROR.append_error_message(stderr), ignore_errors)
+        else:
+            if output.find(DOCKER_PULL_SUCCEEDED.format(IMAGE)) != -1 or \
+               output.find(DOCKER_IMAGE_UP_TO_DATE.format(IMAGE)) != -1:
+                print_pass("Docker pull of '{}'".format(IMAGE))
+            else:
+                _handle_error(DOCKER_PULL_ERROR, ignore_errors)
+
+
+# Get current CLI version
+def _get_cli_version():
+
+    acr_cli_version = "2.2.8"
```

Comment:
Read the version from `src/command_modules/azure-cli-acr/setup.py` for easy maintenance.